robot_calibration/src/sensor_data_generator.cpp

```diff
@@ -0,0 +1,64 @@
+#include "ros/ros.h"
+#include "sensor_msgs/JointState.h"
+#include <cmath>
+#include "yaml-cpp/yaml.h"
+#include <fstream>
+
+class Listener
+{  
+  public:
+    void calculateAndWriteData(const sensor_msgs::JointState::ConstPtr& msg);
+    std::string timestamp = std::to_string(ros::Time::now().toSec());
+};
+
+
+void Listener::calculateAndWriteData(const sensor_msgs::JointState::ConstPtr& msg)
+{
+  double angles[2];
+  angles[0] = msg->position[0];
+  angles[1] = msg->position[1];
+
+  double link_1_length, link_2_length;
+  link_1_length = 1.8;
+  link_2_length = 1.0;
+
+  double ee_position[] = {(link_1_length * cos(angles[0])) + (link_2_length * cos(angles[0] + angles[1])) , (link_1_length * sin(angles[0])) + (link_2_length * sin(angles[0] + angles[1]))}; 
+  
+  angles[0] += 0.044; //theta1 offset
+  angles[1] += 0.013; //theta2 offset
+
+  YAML::Emitter yaml_out_stream;
+  std::ofstream outfile;
+
+  yaml_out_stream << YAML::BeginMap;
+  yaml_out_stream << YAML::Key  << "Angles";
+  yaml_out_stream << YAML::Value;
+  yaml_out_stream << YAML::Flow;
+  yaml_out_stream << YAML::BeginSeq << angles[0] << angles[1] << YAML::EndSeq;
```

Comment:
Right now this is writing to the file _every_ time a message is published to the topic, so the file is getting very large very quickly. You should keep track of the last EE Pos and angles (store in member variables) and only write to the file if there is a difference in value. This will limit the size of the file, and get rid of a lot of duplicate data that adds nothing useful.